pymeasure/instruments/rohdeschwarz/fsl.py

```diff
@@ -253,25 +253,73 @@ def zoom(self, value):
     # Channels -----------------------------------------------------------------
 
     def create_channel(self, channel_type, channel_name):
+        """Create a new channel
+
+        Args:
+            channel_type (string): PNOISE or SANALYZER
+            channel_name (string): Phase Noise, Spectrum analyser, etc
+        """
+
         strict_discrete_set(channel_type, ["PNOISE", "SANALYZER"])
         self.write(f"INST:CRE:NEW {channel_type}, '{channel_name}'")
 
-    def show_channels(self):
-        raw_list = self.values("INST:LIST?")
-        processed_dict = self._channel_list_to_dict(raw_list)
-        return processed_dict
+    def _channel_list_to_dict(raw):
+        """Create a dictionary of open channels and their types
+        Args:
+            raw (list): list of channel names and channel types
+
+        Returns:
+            dictionary : dictionary with channel_name : channel_type
+        """
+        d = {
+            set_keys.strip("'"): set_values.strip("'")
+            for (set_values, set_keys) in zip(raw[0::2], raw[1::2])
+        }
+
+        return d
+
+    show_channels = Instrument.measurement(
+        "INST:LIST?",
+        "To print a dictionary of open channels and their types",
+        get_process=_channel_list_to_dict,
+    )
 
     def delete_channel(self, channel_name):
-        channels = self.show_channels()
+        """Deletes a channel
+
+        Args:
```

Comment:
As stated in another comment, have a look at the docstring style used here: https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html

Reply:
Resolved via [9d73b1d](https://github.com/bleykauf/pymeasure/pull/4/commits/9d73b1d02c1df0a713dcb7e6d7fef6921f07e2e9)